stratum/hal/lib/p4/p4_info_manager.cc

```diff
@@ -164,6 +178,16 @@ P4InfoManager::FindRegisterByName(const std::string& register_name) const {
   return register_map_.FindByName(register_name);
 }
 
+::util::StatusOr<const std::string>
```

Comment:
Not sure if string is the best type to use here. Maybe the protobuf enum is better suited?

Reply:
I was thinking of that too. I used string because P4InfoManager is currently using strings resource type (in the constructor).

Reply:
Ideally using ::p4::v1::Entity::* is a better option. But again, I am using the string convention already existing. If we want to revamp the existing code, I can do that.

Reply:
I tend to agree. Better to change the convention in a separate commit.

Reply:
I am going to merge this change since I have QA waiting for these fixes. I will address the change to Entity in a coming PR.